etc/calico/confd/templates/bird.cfg.template

```diff
@@ -1,3 +1,33 @@
+function apply_communities ()
+{
+  {{- $prefix_advertisements_key := ""}}
+  {{- $node_prefix_advertisements_key := printf "/host/%s/prefix_advertisements" (getenv "NODENAME")}}
+  {{- if exists $node_prefix_advertisements_key}}  
+    {{- $prefix_advertisements_key = $node_prefix_advertisements_key}}
+  {{- else if exists "/global/prefix_advertisements"}}
+    {{- $prefix_advertisements_key = "/global/prefix_advertisements"}}
+  {{- end}}
+  {{- if ne "" $prefix_advertisements_key}}
+    {{- range gets $prefix_advertisements_key}}
```

Comment:
yes, it is a single key, but the value is an array e.g, `[{"cidr":"x.x.x.x/x","communities":["100:6532","101:6532"]},{"cidr":"y.y.y.y/y","communities":["101:6532"]}]`, that's the reason for using `range`, `gets` functions.

I went with single key for entire prefixAdvertisement instead of one key for each object inside `spec.prefixAdvertisements` because it will be complex to compute with unique sub-key names, sub-key has to be a combination of cidr and all the communities in it, in case if the resource looks something like
```yaml
spec:
  prefixAdvertisements: 
    - cidr: x.x.x.x/x
      communities:
      - 100:6532  
      - 101:6532  
    - cidr: x.x.x.x/x
      communities:
      - 101:6532:50  
    - cidr: y.x.x.x/x
      communities:
      - 101:6532:50  
```

Reply:
I'm happy with the internal JSON structure - that seems a very good choice to me.

I still think it would be more conventional to use `exists` to check for the existence of the key, and then `getv` to get its value.  But it's not a big deal, so please consider this as just an idea.